tests/job_tests/returnn/test_search.py

```diff
@@ -0,0 +1,37 @@
+import tempfile
+from sisyphus import setup_path
+
+from i6_core.returnn.search import SearchBPEtoWordsJob
+from i6_core.util import check_file_sha256_checksum, compute_file_sha256_checksum
+
+Path = setup_path(__package__)
+
+
+def test_search_bpe_to_words_single():
+    search_out = Path("files/search_out_single")
+    reference_word_search_results = Path("files/word_search_results_single.py")
+    bpe_to_words_job = SearchBPEtoWordsJob(search_out)
+    bpe_to_words_job.out_word_search_results = Path(tempfile.mktemp(".py"))
+    bpe_to_words_job.run()
+
+    reference_checksum = compute_file_sha256_checksum(
+        reference_word_search_results.get_path()
+    )
```

Comment:
Do we actually need the reference file if we ever only compare against the checksum?

Probably better yes, so we can debug easier if the test fails.

Reply:
Follow up: Do we always need to calculate the checksum each time we test? Since I assume this file does not change wouldn't it be easier to calculate the checksum once and then only compare to that? 
Or is this too prone to errors?

Reply:
The files should be ideally quite small (I will also reduce the file size used here for the test), so calculating the checksum is not expensive. I think it is safer like this, because if you only have the checksum and not the file you can easily make errors. Of course we could include the file in the git and still use a fixed checksum, but I think it is not worth saving that little time on the checks.

Reply:
I agree. Also, if you have the file in the git but still use a fixed checksum, you would need to manually change it if the file ever changes (should usually not happen but who knows...), which can easily be overlooked.

Reply:
If the computation is fast I dont see a reason not to do it, agreed.  :+1: 

We can still keep this in mind though if there are any tests with larger files, but thats for the future.